### PR TITLE
feat(store-indexer): add store indexer service

### PR DESCRIPTION
**Pull request by holic**
pulled out of #1113

### CODE REVIEW

File: packages/store-indexer/bin/sqlite-indexer.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+// TODO: find a better intersection type between sql.js and better-sqlite3 instead of casting here
+const database = drizzle(new Database(env.SQLITE_FILENAME), {
+  // logger: new DefaultLogger(),
+}) as any as BaseSQLiteDatabase<"sync", void>;
+
+let startBlock = env.START_BLOCK;
+try {
+  const currentChainStates = database.select().from(chainState).where(eq(chainState.chainId, chain.id)).all();
+  // TODO: figure out if TS offers an option to turn on `undefined` as a possible outcome of getting an item from an array by index
```
Comment (holic) [inline packages/store-indexer/bin/sqlite-indexer.ts:52]: it does (`noUncheckedIndexedAccess`) but it seems to trickle downstream and reveals a bunch of errors we can't really fix (e.g. in react packages)

File: packages/store-indexer/package.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "@latticexyz/store-indexer",
+  "version": "1.42.0",
```
Comment (alvrs) [inline packages/store-indexer/package.json:3]:
```suggestion
  "version": "2.0.0-next.0",
```

File: packages/store-indexer/src/sqlite/createIndexer.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { debug } from "../debug";
+
+type CreateIndexerOptions = {
+  database: BaseSQLiteDatabase<"sync", any>;
+  publicClient: PublicClient<Transport, Chain>;
+  startBlock: bigint;
+  maxBlockRange: bigint;
+};
+
+export function createIndexer({ database, publicClient, startBlock, maxBlockRange }: CreateIndexerOptions): void {
```
Comment (alvrs) [inline packages/store-indexer/src/sqlite/createIndexer.ts:22]: could we add some tsdoc style code documentation to this function? (ideally for all functions exported by the package)

File: packages/store-indexer/src/sqlite/createStorageAdapter.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { eq } from "drizzle-orm";
+import { BaseSQLiteDatabase } from "drizzle-orm/sqlite-core";
+import { createSqliteTable, chainState, getTables } from "@latticexyz/store-sync/sqlite";
+import { StorageAdapter } from "@latticexyz/store-sync/trpc-indexer";
+import { debug } from "../debug";
+
+export async function createStorageAdapter(database: BaseSQLiteDatabase<"sync", any>): Promise<StorageAdapter> {
```
Comment (alvrs) [inline packages/store-indexer/src/sqlite/createStorageAdapter.ts:7]: let's add some tsdoc style code docs here too

File: packages/store-sync/src/trpc-indexer/createIndexerClient.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { createTRPCProxyClient, httpBatchLink, CreateTRPCProxyClient } from "@trpc/client";
+import superjson from "superjson";
+import type { AppRouter } from "./createAppRouter";
+
+type CreateIndexerClientOptions = {
+  url: string;
+};
+
+export function createIndexerClient({ url }: CreateIndexerClientOptions): CreateTRPCProxyClient<AppRouter> {
```
Comment (alvrs) [inline packages/store-sync/src/trpc-indexer/createIndexerClient.ts:9]: let's add a tsdoc code comment here too

**Comment by holic** [comment]
TS issue is coming from https://github.com/drizzle-team/drizzle-orm/issues/682

File: packages/store-indexer/package.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    "build:js": "tsup",
+    "clean": "pnpm run clean:js",
+    "clean:js": "rimraf dist",
+    "dev": "tsup --watch",
+    "lint": "eslint .",
+    "start": "tsx bin/sqlite-indexer",
+    "start:local": "SQLITE_FILENAME=anvil.db CHAIN_ID=31337 pnpm start",
+    "start:testnet": "SQLITE_FILENAME=testnet.db CHAIN_ID=4242 START_BLOCK=19037160 pnpm start",
+    "start:testnet2": "SQLITE_FILENAME=testnet2.db CHAIN_ID=4243 pnpm start",
+    "test": "tsc --noEmit --skipLibCheck"
```
Comment (holic) [inline packages/store-indexer/package.json:27]: We should remove the `skipLibCheck`  flag once https://github.com/drizzle-team/drizzle-orm/issues/682 is fixed.

**Pull request merged**